src/jobs/java_maven_build_test.yml

```diff
@@ -0,0 +1,48 @@
+description: >
+  Build and test java app using maven; update circleCI caches and workspace.
+  Supports app-module from monorepo as well as single app from single-app-repo.
+
+parameters:
+  app_path:
+    description: "Path of maven module for the app, or \".\" for single-app-repo"
+    type: string
+    default: "."
+  maven_cache_key_prefix:
+    description: "Prefix for the maven artifacts cache-key (used in combination with checksum of pom.xml); no caching with blank prefix"
+    type: string
+    default: ""
+  executor:
+    description: "Executor for the build (java_builder_docker is more lightweight, but some tests leveraging testcontainers etc require the java_builder_vm)"
+    type: enum
+    enum: [ "java_builder_docker", "java_builder_vm"]
+    default: "java_builder_docker"
+  java_builder_image:
+    description: "Image to use for the executor"
+    type: string
+    default: "cimg/openjdk:11.0"
+  java_builder_docker_hub_username:
+    description: "Docker hub credentials to download the executor image"
+    type: string
+    default: $DOCKER_HUB_USERNAME
+  java_builder_docker_hub_password:
+    description: "Docker hub credentials to download the executor image"
+    type: string
+    default: $DOCKER_HUB_PASSWORD
+
+executor:
+  name: << parameters.executor >>
+  image: << parameters.java_builder_image >>
+  docker_hub_username: << parameters.java_builder_docker_hub_username >>
+  docker_hub_password: << parameters.java_builder_docker_hub_password >>
```

Comment:
to me this seems like we always use docker and never a machine based executor. am i wrong?

Reply:
no. the executor is selected with the executor param (by name)